JS/Lesson34/onur/my-app/src/components/NewRecipe/index.jsx

```diff
@@ -22,56 +23,195 @@ export const NewRecipe = () => {
 
 	const strCategory = watch('strCategory');
 
-	const onSubmit = (data) => console.log(data);
+
+
+
+	const onSubmit = (data) => {
+
+		if (data.strDrinkAlternate.trim() === '') {
+			data.strDrinkAlternate = null;
```

Comment:
I see that you're trying to modify the data that is coming from the form, if the value of input is empty (or spaces).  React-hook-form does not actually allow you to modify the incoming form data like this so this operation does not make any sence. 

Instead, you could add a pattern match for required input fields: /\S+/ regex should match any character exclusive white spaces so that your input will only be valid if there is an actual word or other character added.